src/charm.py

```diff
@@ -554,6 +643,132 @@ def _update(self, event):
 
         self.unit.status = MaintenanceStatus("replanning application")
 
+    # Helpers for frontend TLS
+    def _relation_created(self, relation_name: str) -> bool:
+        return bool(self.model.relations.get(relation_name))
+
+    def _certificate_is_available(self) -> bool:
+        cert, key = self.certificates.get_assigned_certificate(
+            certificate_request=self._get_certificate_request_attributes()
+        )
+        return bool(cert and key)
+
+    def _get_certificate_request_attributes(self) -> CertificateRequestAttributes:
+        """Return the attributes of the certificate this charm will request."""
+        # Generate SANS IP - use the unit ip in case any clients try to reach
+        # the frontend server using it
+        unit_hostname = socket.getfqdn()
+        sans_ip = frozenset((socket.gethostbyname(unit_hostname),))
+
+        # Generate SANS_DNS - set to the k8s service name. If integrated with traefik,
+        # use the URL it provides.
+        k8s_svc_dns = f"{self.app.name}.{self.model.name}.svc.cluster.local"
+        sans_dns = frozenset([k8s_svc_dns, unit_hostname])
```

Comment:
A similar point here, most TLS providers won't sign a certificate for `<something>`.svc.cluster.local`.